Makefile

```diff
@@ -1,19 +1,34 @@
+TARGETS = naive cuda
+
+.PHONY: default all debug clean $(TARGETS)
+
 CXXFLAGS := -O2 -std=c++17
 CXX := g++
 
 CUDAFLAGS := -O2 -std=c++14 --expt-relaxed-constexpr
-NVCC:= nvcc
+NVCC := nvcc
 
 default: naive
 
-naive:
+all: $(TARGETS)
+
+debug: $(TARGETS:%=debug-%)
+
+clean:
+	rm -f $(TARGETS:%=main-%) $(TARGETS:%=debug-%)
+
+naive: main-naive
+
+main-naive: main.cc rawtodigi_naive.h
```

Comment:
In principle `input.h output.h pixelgpudetails.h` should be added here and the other targets as well.

I wished this project would stay simple-enough that just always compiling would be good-enough so that the proper dependency management could be ignored. I won't object though adding the dependencies.